Automattic/jetpack
# SSO: remove 'wpcom_user_id' user metadata verification

## Proposed changes:
* SSO: remove `wpcom_user_id` user metadata verification
* SSO: add user token verification via WPCOM API

### Other information:


## Jetpack product discussion
p9dueE-70y-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect a Jetpack site.
2. Go to "Jetpack -> Settings" and enable "WordPress.com login".
3. Log out of the Jetpack site, log in again using via "Log in with WordPress.com"
4. Confirm you get logged in with no issues.
5. Set an invalid user token via Jetpack Debug plugin.
6. Logout, try to login again via SSO.
7. Confirm that you see the error message:
![sso](https://user-images.githubusercontent.com/1341249/228962233-7b607631-2d61-42f0-bab1-727674736622.png)
